tools/c7n_azure/tests/test_firewall_actions.py

```diff
@@ -25,92 +25,56 @@ def test_build_bypass_rules(self):
-                'networkAcls': {
-                    'virtualNetworkRules': [
-                        {'id': 'Hello'}, {'id': 'World'}
-                    ]
-                }
-            }
-        }
-
-        action = StorageSetNetworkRulesAction(data)
+        action = StorageSetFirewallAction(data)
         action.append = False
-        rules = action._build_vnet_rules(resource, data['virtual-network-rules'])
+        rules = action._build_vnet_rules(['Hello', 'World'], data['virtual-network-rules'])
         self.assertEqual(sorted(['id1', 'id2']), sorted(rules))
 
         action.append = True
-        rules = action._build_vnet_rules(resource, data['virtual-network-rules'])
+        rules = action._build_vnet_rules(['Hello', 'World'], data['virtual-network-rules'])
         self.assertEqual(sorted(['id1', 'id2', 'Hello', 'World']), sorted(rules))
 
     def test_build_ip_rules(self):
         data = {
             'ip-rules': ['1.1.1.1', '6.0.0.0/16']
         }
 
-        resource = {
-            'properties': {
-                'networkAcls': {
-                    'ipRules': [
-                        {'value': '1.1.1.1'}, {'value': '8.0.0.0/12'}]
-                }
-            }
-        }
-
-        action = StorageSetNetworkRulesAction(data)
+        action = StorageSetFirewallAction(data)
         action.append = False
-        rules = action._build_ip_rules(resource, data['ip-rules'])
+        rules = action._build_ip_rules(['1.1.1.1', '8.0.0.0/12'], data['ip-rules'])
         self.assertEqual(sorted(['1.1.1.1', '6.0.0.0/16']), sorted(rules))
 
         action.append = True
-        rules = action._build_ip_rules(resource, data['ip-rules'])
+        rules = action._build_ip_rules(['1.1.1.1', '8.0.0.0/12'], data['ip-rules'])
         self.assertEqual(sorted(['1.1.1.1', '6.0.0.0/16', '8.0.0.0/12']), sorted(rules))
 
     def test_build_ip_rules_alias(self):
         data = {
             'ip-rules': ['ServiceTags.ApiManagement.WestUS', '6.0.0.0/16']
         }
 
-        resource = {
-            'properties': {
-                'networkAcls': {
-                    'ipRules': [
-                        {'value': '1.1.1.1'}, {'value': '8.0.0.0/12'}]
-                }
-            }
-        }
-
-        action = StorageSetNetworkRulesAction(data)
+        action = StorageSetFirewallAction(data)
         action.append = False
-        rules = action._build_ip_rules(resource, data['ip-rules'])
+        rules = action._build_ip_rules(['1.1.1.1', '8.0.0.0/12'], data['ip-rules'])
         self.assertIn('6.0.0.0/16', rules)
         self.assertEqual(4, len(rules))
 
         # With append we expect all our specified values + others from the service tag.
         action.append = True
-        rules = action._build_ip_rules(resource, data['ip-rules'])
+        rules = action._build_ip_rules(['1.1.1.1', '8.0.0.0/12'], data['ip-rules'])
         self.assertTrue({'6.0.0.0/16', '1.1.1.1', '8.0.0.0/12'} <= set(rules))
```

Comment:
assertEqual here?

Reply:
This is a subset test - I am just confirming expected is contained in actual as there will also be other IP's there from the service tag.